subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -2303,6 +2502,41 @@ void lwm2m_udp_receive(struct lwm2m_ctx *client_ctx, uint8_t *buf, uint16_t buf_
 			}
 		}
 
+#if defined(CONFIG_LWM2M_COAP_BLOCK_TRANSFER)
+		if (coap_header_get_code(&response) == COAP_RESPONSE_CODE_CONTINUE) {
+
+			r = coap_get_block1_option(&response, &more_blocks, &block_num);
+			if (r < 0) {
+				LOG_ERR("Missing block1 option in response with continue");
+			}
+			if (r != CONFIG_LWM2M_COAP_BLOCK_SIZE) {
+				LOG_WRN("Server requests different block size: ignore");
+			}
+			if (!more_blocks) {
+				lwm2m_reset_message(msg, true);
+				LOG_ERR("Missing more flag in response with continue");
+				return;
+			}
+
+			r = build_msg_block_for_send(msg, block_num + 1);
+			if (r < 0) {
+				lwm2m_reset_message(msg, true);
+				LOG_ERR("Unable to build next block of lwm2m message!");
+				return;
+			}
+			r = lwm2m_send_message_async(msg);
+			if (r < 0) {
+				lwm2m_reset_message(msg, true);
+				LOG_ERR("Unable to send next block of lwm2m message!");
+				return;
+			}
+
+			/* skip release as message was reused for new block */
```

Comment:
If you don't reset the fragment message, who releases the reply/pending structs then? A new one is allocated for the next fragment unless I miss something.

Reply:
In `build_msg_block_for_send(..)` we call `lwm2m_reset_message(msg, false)`. There the `pending`and the `reply` pointers of the message are cleared.